includes/Checker/Abstract_Check_Runner.php

```diff
@@ -41,7 +75,60 @@ abstract protected function get_checks_instance();
 	 *
 	 * @return array An array of Check slugs.
 	 */
-	abstract protected function get_check_slugs_to_run();
+	abstract protected function get_check_slugs_param();
+
+	/**
+	 * Sets whether the runner class was initialized early.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function __construct() {
+		$this->initialized_early = ! did_action( 'muplugins_loaded' );
+	}
+
+	/**
+	 * Sets the check slugs to be run.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $check_slugs An array of check slugs to be run.
+	 *
+	 * @throws Exception Thrown if the checks do not match those in the original request.
+	 */
+	public function set_check_slugs( array $check_slugs ) {
+		if ( $this->initialized_early ) {
+			// Compare the check slugs to see if there was an error.
+			if ( $check_slugs !== $this->get_check_slugs_param() ) {
+				throw new Exception(
+					__( 'Invalid checks: The checks to run do not match the original request.', 'plugin-check' )
+				);
+			}
+		}
+
+		$this->check_slugs = $check_slugs;
+	}
+
+	/**
+	 * Sets the plugin slug to be checked.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_slug The plugin slug to be checked.
+	 *
+	 * @throws Exception Thrown if the plugin main file does not match the original request.
+	 */
+	public function set_plugin_slug( $plugin_slug ) {
```

Comment:
Currently the logic in this checks against the plugin parameter, which can be either a plugin slug or a plugin basename, so the name of this method, the parameter, and also the internal property name is misleading.

We should probably align with the rest of the new method names here to clarify that this is simply the `plugin` param. In other words, let's call this `set_plugin( $plugin )`, and let's clarify on both the parameter and the `$this->plugin` property declaration that the value can be either a plugin slug or a plugin basename.

This effectively means wherever we use this value we need to run it through `Plugin_Request_Utility::get_plugin_basename_from_input` to sanitize it to a basename. I think that's okay since it only happens in one place anyway.